scripts/src/public/services/orders.ts

```diff
@@ -223,8 +223,9 @@ async function createNormalSpendExternalOrder(sender: User, jwt: ExternalSpendOr
 
 export async function createExternalOrder(jwt: string, user: User, logger: LoggerInstance): Promise<OpenOrder> {
 	const payload = await validateExternalOrderJWT(jwt, user.appUserId, logger);
+	const nonce = payload.nonce || db.Order.DEFAULT_NONCE;
 
-	let order = await db.Order.findBy(payload.offer.id, user.id);
+	let order = await db.Order.findBy({ offerId: payload.offer.id, userId: user.id, nonce });
```

Comment:
There is a bug here that you don't order the Orders by date, so if you failed once the findBy might always return the failed order instead of the latest one.
I think you should add `order by date desc` to `db.Order.findBy` function

Reply:
nice catch.  added